### PR TITLE
[MTSRE-1171] feat: improve initialization for kubectl-package

### PR DESCRIPTION
**Pull request by Ajpantuso**
### Summary

- Sets up uniform initialization for subcommands so that the kubect-package root can inject dependencies
- Sets up `zap` as the logger for `kubectl-package` and adds zap options to the persistent flags
- Repackages the root command under `kubectl-package`

### Change Type

New Feature

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: cmd/kubectl-package/deps/command.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+type defaultRendererFactory struct {
+	logFactory LogFactory
+	scheme     *runtime.Scheme
+}
+
+func (f *defaultRendererFactory) Renderer() treecmd.Renderer {
+	return internalcmd.NewTree(
+		f.scheme,
+		internalcmd.WithLog{
+			Log: f.logFactory.Logger().V(1),
```
Comment (eqrx) [inline cmd/kubectl-package/deps/command.go:61]: Is all output of tree debug output? (zaps DebugLevel lvl)

Comment (Ajpantuso) [reply cmd/kubectl-package/deps/command.go:61]: I'm fine removing the `v(1)`'s for this and the `builder`. It's a hold over from when we weren't using levels. I've set `zap` to default to error so `Info` level will not be shown unless the user asks for it.

Comment (Ajpantuso) [reply cmd/kubectl-package/deps/command.go]: The `V(1)`'s are gone now.

**Review by eqrx — APPROVED** [review]

**Pull request merged**